Error popup on uncaught exceptions

Neui opened the pull request:
Did a while ago, but currently paused but planned to continue somewhat soon. Just in case: This is still Work In Progress (WIP)!

Basically when an uncaught exceptions is being caught (either explicit via a decorator or implicit via a global hook), it'll then display an dialog window with the info and ask the user to report it.

Also, would be a good place to also have an popup when something fails during initialization (which previously would just be silent end error out later due to uninitialized attributes).

Some ideas thrown in:
* Keep an action log of the last x actions to be included
* Maybe also include x lines of logs

Related: #247 but unlikely to have (explicit) integration with such crash reporters, for now.

First shared in the (Matrix) chat, here some (now outdated) screenshots:

![first](https://matrix-client.matrix.org/_matrix/media/r0/download/matrix.org/bCJKTUMliCJkefhlvXYmwbmu)

![second](https://matrix-client.matrix.org/_matrix/media/r0/download/matrix.org/gkUeUmdQGCEBnabjSVeAhEHE)

Neui commented:
I'll un-draft it so it doesn't get stuck on "making it more perfect". It seems to work, so I guess it can be reviewed. The ideas can be sometime implemented later (better create an issue to keep track of it)

diegogangl commented:
Hey, this is awesome! Looks good, just some commented code and a conflict in `application.py`

Neui commented:
Fixed. I also changed so that it doesn't print the exception twice in the console, and added special check for `Gtk.ResponseType.DELETE_EVENT` (to prevent an "unknown response" log message)

diegogangl commented:
Awesome, thanks!
